examples/src/main/scala/com/cloudera/livy/apps/WordCountApp.scala

```diff
@@ -0,0 +1,196 @@
+    * @param port Port that Spark Streaming context has to connect for receiving data
+    * @param outputPath Output path to save the processed data read by the Spark Streaming context
+    */
+  def processStreamingWordCount(host: String, port: Int, outputPath: String): ScalaJobHandle[Unit] = {
+    val handle = scalaClient.submit { context =>
+      context.createStreamingContext(20000)
+      val ssc = context.streamingctx
+      val sqlctx = context.sqlctx
+      val lines = ssc.socketTextStream(host, port, StorageLevel.MEMORY_AND_DISK_SER)
+      val words = lines.filter(filterEmptyConent(_)).flatMap(tokenize(_))
+      words.print()
+      words.foreachRDD { rdd =>
+        import sqlctx.implicits._
+        val df = rdd.toDF("word")
+        df.write.mode("append").json(outputPath)
+      }
+      ssc.start()
+      ssc.awaitTerminationOrTimeout(35000)
+      ssc.stop(false, true)
+    }
+    handle
+  }
+
+  /**
+    * Submits a spark sql job to the livy server
+    * <p/>
+    * The sql context job reads data frames from the given json path and executes
+    * a sql query to get the word with max count on the temp table created with data frames
+    * @param inputPath Input path to the json data containing the words
+    */
+  def getWordWithMostCount(inputPath: String): ScalaJobHandle[Any] = {
+    val handle = scalaClient.submit { context =>
+      val sqlctx = context.sqlctx
+      val rdd = sqlctx.read.json(inputPath)
+      rdd.registerTempTable("words")
+      val result = sqlctx.sql("select word, count(word) as word_count from words " +
+        "group by word order by word_count desc limit 1")
+      if (result != null && result.count() > 0) {
+        result.collect()(0)
+      }
+    }
+    handle
+  }
+
+  private def filterEmptyConent(text: String): Boolean = {
+    if (text == null || text.isEmpty) false else true
+  }
+
+  private def tokenize(text : String) : Array[String] = {
+    text.toLowerCase.replaceAll("[^a-zA-Z0-9\\s]", "").split("\\s+")
+  }
+
+  /**
+    * Main method of the WordCount App
+    * STEPS FOR EXECUTION
+    * 1) 'livy.file.local-dir-whitelist' config has to be set in livy.conf file for the jars to be
+    * added to the session
+    *
+    * 2) Libraries to be added to classpath - livy-api-version-SNAPSHOT.jar,
+    * livy-scala-api-version-SNAPSHOT.jar, livy-client-http-version-SNAPSHOT.jar
+    *
+    * @param args
+    * arg(0) - Livy server url
+    * arg(1) - Output path to save the text read from the stream
+    */
+  def main(args: Array[String]) {
+    if (args.length < 2) {
+      throw new IllegalArgumentException("Number of args is less")
+    }
+    val outputPath = args(1)
+    init(args(0))
+    uploadRelevantJarsForJobExecution()
+    logger.info("Calling processStreamingWordCount")
+    val handle1 = processStreamingWordCount("localhost", 8086, outputPath)
+    Await.result(handle1, 120 second) match {
+      case success: Unit => logger.info("Successfully read data from stream")
```

Comment:
The job should be returning something, right? The most common word. So this should actually be causing a MatchError, since it's returning something that is not `Unit`.